Loosen dependency on typing-extensions

## Explanation

`pypdf` currently depends on `typing_extensions>=3.10`. Is there any reason for this minimum version?

Background: I am currently bound to using 3.7.4.3 to maintain compatibility with some other packages. Running the tests locally with `typing_extensions==3.7.4.3` does not seem to introduce any issues on Python 3.6.15 and Python 3.9.16.

> In any case our CI should capture issues.

This only holds true if `requirements/ci.txt` is updated as well to reflect the new version.

Checking this, it seems to like updating `requirements/ci.in` and `requirements/ci.txt` accordingly for CI seems to work for *pytest* (as already observed in my local tests).

But: Apparently *mypy* made the same (wrong) assumptions in https://github.com/python/mypy/commit/71c5a0e37741a4452803122ef23b76c022fab607 and requires pinning to a maximum of `mypy==0.910` if we want to relax the *typing_extensions* version as proposed. This reports some "new" *mypy* issues:

```none
pypdf/generic/_data_structures.py:869: error: Incompatible types in assignment (expression has type "None", variable has type "ArrayObject")  [assignment]
pypdf/filters.py:785: error: Argument "formats" to "open" has incompatible type "Tuple[str]"; expected "Union[List[str], Tuple[str, Any], None]"  [arg-type]
pypdf/filters.py:867: error: Argument "formats" to "open" has incompatible type "Tuple[str]"; expected "Union[List[str], Tuple[str, Any], None]"  [arg-type]
pypdf/_writer.py:[12](https://github.com/stefan6419846/pypdf/actions/runs/5495158775/jobs/10014434259#step:12:13)83: error: Argument 1 to "FileIO" has incompatible type "Union[Path, str]"; expected "Union[str, bytes, int]"  [arg-type]
pypdf/_writer.py:2686: error: Argument 1 to "FileIO" has incompatible type "Union[Path, str]"; expected "Union[str, bytes, int]"  [arg-type]
pypdf/_merger.py:254: error: Argument 1 to "FileIO" has incompatible type "Union[Path, str]"; expected "Union[str, bytes, int]"  [arg-type]
Found 6 errors in 4 files (checked 33 source files)
```

Given this, I am not sure whether it is desired on the PyPDF side to officially support this older *typing_extensions* version, as this basically requires us to rewrite/change the GitHub workflows to ensure compatibility accordingly:

* One complete run for each supported Python version with `typing_extensions==3.7.4.2`, but with *mypy* checks disabled.
* One complete run for each supported Python version with `typing_extensions==3.10` with the *mypy* checks enabled.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -28,7 +28,7 @@ classifiers = [
     "Typing :: Typed",
 ]
 dependencies = [
-    "typing_extensions >= 3.10.0.0; python_version < '3.10'",
+    "typing_extensions >= 3.7.4.2; python_version < '3.10'",
     "dataclasses; python_version < '3.7'",
 ]
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -28,7 +28,7 @@ classifiers = [
     "Typing :: Typed",
 ]
 dependencies = [
-    "typing_extensions >= 3.7.4.2; python_version < '3.10'",
+    "typing_extensions >= 3.7.4.3; python_version < '3.10'",
     "dataclasses; python_version < '3.7'",
 ]
 

